### DISCUSSION TITLE
Trying to figure out if Control has a binding

### DISCUSSION THREAD

**Discussion by vb2ae** [AvaloniaUI/Avalonia#8290] [Q&A] [answered]
In WPF you can use BindingOperations.GetBindingBase to figure out if a control already has a DependencyProperty.  Is there a way to figure out if a control already has a binding to an existing property?  I see there is GetBindingObservable but it always returns a new AvaloniaPropertyBindingObservable

        public static IObservable<BindingValue<object?>> GetBindingObservable(
            this IAvaloniaObject o,
            AvaloniaProperty property)
        {
            return new AvaloniaPropertyBindingObservable<object?>(
                o ?? throw new ArgumentNullException(nameof(o)),
                property ?? throw new ArgumentNullException(nameof(property)));
        }

**Comment by maxkatz6** [comment, marked as answer]
> if a control already has a binding to an existing property

I don't think it's possible. Technically single property might have multiple bindings, but I think technically it would be possible to get one with highest priority. 

> to figure out if a control already has a DependencyProperty

Not sure if I understand. You want to know, if control has a value for specific property (attached or direct/styled), right?
In this case `avaloniaObject.IsSet(AvaloniaProperty)` should work for you.

Or maybe you want to know if a property is registered for a control? In this case there is a AvaloniaPropertyRegistry class to get them per type.
